src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs

```diff
@@ -220,7 +223,16 @@ private void AddTemplateOptionsToCommand(CliTemplateInfo templateInfo)
             //TODO: handle errors
             var parametersWithAliasAssignments = AliasAssignmentCoordinator.AssignAliasesForParameter(parameters, initiallyTakenAliases);
 
-            foreach ((CliTemplateParameter parameter, IReadOnlyList<string> aliases, IReadOnlyList<string> errors) in parametersWithAliasAssignments)
+            IReadOnlyDictionary<CliTemplateParameter, IReadOnlyList<string>> errors = parametersWithAliasAssignments
+                .Where(p => p.Errors.Any())
+                .ToDictionary(p => p.Parameter, p => p.Errors);
+
+            if (errors.Any())
+            {
+                throw new InvalidTemplateParametersException(templateInfo, errors);
+            }
+
+            foreach ((CliTemplateParameter parameter, IReadOnlyList<string> aliases, IReadOnlyList<string> _) in parametersWithAliasAssignments)
```

Comment:
It seems that we don't need the dictionary unless there is an error.
```suggestion
            IReadOnlyDictionary<CliTemplateParameter, IReadOnlyList<string>> errors = parametersWithAliasAssignments
                .Where(p => p.Errors.Any());

            if (errors.Any())
            {
                throw new InvalidTemplateParametersException(templateInfo, errors.ToDictionary(p => p.Parameter, p => p.Errors));
            }

            foreach ((CliTemplateParameter parameter, IReadOnlyList<string> aliases, IReadOnlyList<string> _) in parametersWithAliasAssignments)
```

Reply:
Agreed, refactored.